feat: LEAP-883: Custom Scripts

Gondragos opened the pull request:
- add `idAttr` parameter into `View` tag to make easier to work with custom styles and so on
- add `lsfInit` and `beforeLsfDestroy` events into data manager for better control of the life cycle

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
`DataManager`, `View`

Gondragos committed:
feat: LEAP-883: Custom Scripts

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9068862105

Gondragos committed:
Prepare SDKs for using LSF related plugins

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9070111296

Gondragos committed:
Fix arguments passing in lsfInit and beforeLsfDestroy events

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9070584265

Gondragos committed:
Merge branch 'develop' into fb-leap-883/custom-scripts

# Conflicts:
#	web/dist/apps/labelstudio/version.json
#	web/dist/libs/datamanager/version.json
#	web/dist/libs/editor/version.json

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9116654133

Gondragos committed:
Add idAttr into View and fix documentation

Add ability to set id on div rendered by View to make it easy to use with for example custom styles

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9160874538

Gondragos committed:
Merge branch 'develop' into fb-leap-883/custom-scripts

# Conflicts:
#	web/dist/apps/labelstudio/version.json
#	web/dist/libs/datamanager/main.js
#	web/dist/libs/datamanager/main.js.map
#	web/dist/libs/datamanager/version.json
#	web/dist/libs/editor/version.json

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9175652845

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9193940641)
> Successfully merged:  create mode 100644 label_studio/data_manager/actions/cache_labels.py

hlomzik committed:
Merge branch 'develop' into 'fb-leap-883/custom-scripts'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9193940641

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9193977328

Commit:
feat: LEAP-1060: Add event to control annotations submit (#5891)

This PR improves the way of saving annotations (submit and update) with
allowing to prevent submitting in some cases such as validation.

- feat: LEAP-1060: Add event to allow control annotations submitting
- Fix behavior in case of absence of event listeners
- Fix submitting without event handlers for beforeSaveAnnotation

### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated (for bug
fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and
coherent (checked for spelling/grammatical errors, commented out code,
debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug
fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
A missed useful event.




### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
`Annotation`, `Validation`

---------

Co-authored-by: robot-ci-heartex <robot-ci-heartex@users.noreply.github.com>
Co-authored-by: hlomzik <hlomzik@gmail.com>

Gondragos committed:
Fix initialization in simple init case

Gondragos committed:
Merge branch 'develop' into fb-leap-883/custom-scripts

# Conflicts:
#	web/dist/apps/labelstudio/version.json
#	web/dist/libs/datamanager/main.js
#	web/dist/libs/datamanager/version.json
#	web/dist/libs/editor/main.js
#	web/dist/libs/editor/main.js.map
#	web/dist/libs/editor/version.json

Gondragos committed:
Add feature flag

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9256198341

Gondragos committed:
Revert history fix

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9256286613

Gondragos committed:
Add "selectHistory" event

Gondragos committed:
Merge branch 'fb-leap-883/custom-scripts' of ssh://github.com/heartexlabs/label-studio into fb-leap-883/custom-scripts

Gondragos committed:
Merge branch 'develop' into fb-leap-883/custom-scripts

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9269695511

Gondragos committed:
Add beforeSaveAnnotation event into accepting and rejecting actions

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9270510702
